### PR TITLE
Add configs for ETCD druid for local gardener example

### PR DESCRIPTION
**Pull request by abdasgupta**
**How to categorize this PR?**

/area testing
/kind test

**What this PR does / why we need it**:
This PR adds `etcdConfig` section for `example/gardener-local/gardenlet/values.yaml` so that ETCD Druid can be configured easily in a local setup.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```noteworthy developer
Developers can easily configure ETCD Druid for local setup using the gardenlet config values in example directory
```

### CODE REVIEW

**Review by shafeeqes — COMMENTED** [review]
/kind enhancement
/approve
/cc @aaronfern

**Comment by shafeeqes** [comment]
Don't we need to add
https://github.com/gardener/gardener/blob/2e4617c3de1a4491a6287788266d45a7fc8044e2/charts/gardener/gardenlet/values.yaml#L188-L190
too?

**Comment by abdasgupta** [comment]
> Don't we need to add
> 
> https://github.com/gardener/gardener/blob/2e4617c3de1a4491a6287788266d45a7fc8044e2/charts/gardener/gardenlet/values.yaml#L188-L190
> 
> 
> too?

I made the requested changes

File: example/gardener-local/gardenlet/values.yaml [hunk 1]
```diff
@@ -59,6 +59,19 @@ config:
...
+    custodianController:
+      workers: 3
+    backupCompactionController:
+      workers: 3
+      enableBackupCompaction: true
+      eventsThreshold: 10000
+      activeDeadlineDuration: "3h"
+    backupLeaderElection:
+      reelectionPeriod: 5s
+      etcdConnectionTimeout: 5s
```
Comment (ialidzhikov) [inline example/gardener-local/gardenlet/values.yaml]: I think this file is used for e2e tests and local setup. Can you elaborate why the default values don't work and we have to specify the above values?

Comment (ialidzhikov) [reply example/gardener-local/gardenlet/values.yaml]: `etcdController.workers` has default value 50, `custodianController.workers` has default value 10, `backupCompactionController. workers` has already default value 3, `backupCompactionController. enableBackupCompaction` has default value false, `backupCompactionController.eventsThreshold` has default value 1000000, I don't see any defaulting in g/g for `backupCompactionController.activeDeadlineDuration` but its doc string says that it default to 3h. For `backupLeaderElection` fields I don't see any defaulting in g/g.

https://github.com/gardener/gardener/blob/2e4617c3de1a4491a6287788266d45a7fc8044e2/pkg/gardenlet/apis/config/v1alpha1/defaults.go#L474-L500

From the context I have, I guess there is need only to set `backupCompactionController.enableBackupCompaction` and `backupCompactionController.eventsThreshold`. I guess it does not harm to have high values for the `workers` values. 

I am writing all of this as the pattern that seems to be followed in this file is to specify only the values which are different from the defaults.

For example, only 2 values are specified for gardenlet controllers https://github.com/gardener/gardener/blob/f81703bda20f5be17356b1d2f4cd9c740b3a9ea2/example/gardener-local/gardenlet/values.yaml#L32-L36 while while there are actually many controllers with different kind of configs - https://github.com/gardener/gardener/blob/2e4617c3de1a4491a6287788266d45a7fc8044e2/example/20-componentconfig-gardenlet.yaml#L17-L89.

---

I would also ask why we miss defaulting for the gardenlet component config for fields like `backupCompactionController.activeDeadlineDuration` and `backupLeaderElection`? Is there a specific reason?

Comment (abdasgupta) [reply example/gardener-local/gardenlet/values.yaml]: I raised this PR as it's helpful for a new user to easily set druid config. Yes, the values can have same values as the default value but if somebody wants to test something, they can easily change the values in this file and test the druid. I already got such requests from people to show how to tweak the `eventhreshold` value for compaction in local setup. I thought, this PR would be good to have

Comment (aaronfern) [reply example/gardener-local/gardenlet/values.yaml]: I'd agree it's fair not to set values that aren't needed here. We don't need to include `etcdController` and `custodianController` in the config, but we can stick with `backupCompactionController`

Comment (aaronfern) [reply example/gardener-local/gardenlet/values.yaml]: > I would also ask why we miss defaulting for the gardenlet component config for fields like backupCompactionController.activeDeadlineDuration and backupLeaderElection? Is there a specific reason?

@ialidzhikov from [here](https://github.com/gardener/gardener/blob/master/pkg/component/etcd/bootstrap.go#L400-L401) I can see that we don't want to set the `--active-deadline-duration` flag for etcd druid if `backupCompactionController.activeDeadlineDuration` is not set. So ig it makes sense not to have a default value

I didn't check the code for `backupLeaderElection`, however I would assume it's the same reason

Comment (ialidzhikov) [reply example/gardener-local/gardenlet/values.yaml]: > I raised this PR as it's helpful for a new user to easily set druid config. Yes, the values can have same values as the default value but if somebody wants to test something, they can easily change the values in this file and test the druid. I already got such requests from people to show how to tweak the `eventhreshold` value for compaction in local setup. I thought, this PR would be good to have

Sure. In https://github.com/gardener/gardener/pull/8331#discussion_r1288365149 I tried to share that it is fine to include settings such as `backupCompactionController.enableBackupCompaction` and `backupCompactionController.eventsThreshold`. But including workers count and leader election config - I don't recall a case in local setup I needed to tweak such settings for any controller - the defaults usually worked.
We also have extensive examples in [`example/20-componentconfig-gardenlet.yaml`](https://github.com/gardener/gardener/blob/master/example/20-componentconfig-gardenlet.yaml) where every possible field should be present.

tl;dr: I would vote to only include the values that really need to be configured for local setup (for example default doesn't work or doesn't make sense) and values which are different from field defaults.
